### ISSUE TITLE
S3 pre-signed URLs do not expire

### ISSUE THREAD

**Issue by sola91** [localstack/localstack#1685]
I am using Localstack S3 in order to test that a link correctly expires after a given TimeToLive.
Here is a short scala snippet of what I am doing:

``` 
def createTemporaryLink(s3: AmazonS3, s3ObjectPath: String, temporaryLinkTtl: FiniteDuration): String = {
    val createdAt   = ZonedDateTime.now(ZoneOffset.UTC)
    val expiresAt   = Date.from(createdAt.plus(temporaryLinkTtl.asJava).toInstant)
    val amazonS3Uri = new AmazonS3URI(s3ObjectPath)
    val generatePresignedUrlRequest: GeneratePresignedUrlRequest =
      new GeneratePresignedUrlRequest(amazonS3Uri.getBucket, amazonS3Uri.getKey)
        .withMethod(HttpMethod.GET)
        .withExpiration(expiresAt)
    s3.generatePresignedUrl(generatePresignedUrlRequest).toString
  }
```
The code works and I get a valid link and I can even run an HTTP request and get the content of the file. However, the issue is that I can access the content of the s3 object even when the expirationTime has passed

**Comment by jrr** [comment]
Can you repro this with the AWS CLI? I seem to recall seeing pre-signed URLs expire last time I tried.

**Comment by whummer** [comment]
Thanks for reporting @sola91 @jrr . This should be fixed in #2333 - can you please give it a try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks!
